content/en/tracing/profiler/intro_to_profiling.md

```diff
@@ -0,0 +1,193 @@
+---
+title: Intro to Profiling
+kind: Documentation
+---
+
+You may have heard that profiling can make your services faster, cheaper, and more reliable, but if you haven't used a profiler before, it can seem like a dark art.
+
+In this guide, we'll explain profiling, then profile a sample service that has a performance problem and see how we can use Continuous Profiler to understand and fix the problem!
+
+## What is Profiling?
+
+A profiler shows how much "work" each line of code is doing by collecting data about the program as it's running. For example, infrastructure monitoring might show that your app servers are using 80% of their CPU but you may not know why. Profiling could then show you that function `doSomeWork` is using 48% of the CPU, function `renderGraph` is using another 19%, and so on. This is important when working on performance problems because many programs spend a lot of time in very few places and it's often not obvious which places those are. Guessing at which parts of a program to optimize can often cause engineers to spend a lot of time and not get much in the way of results. By using a profiler, we find exactly which parts of the code we should optimize to get the most bang for our buck.
+
+If you've used an APM tool, you might think of profiling like a "deeper" tracer that provides a very fine grained view of your code without needing any instrumentation.
+
+Datadog Continuous Profiler can track various types of "work", including CPU usage, amount and types of objects being allocated in memory, time spent waiting to acquire locks, amount of network or file I/O, and more! The profile types available depend on the language being profiled.
+
+We've got an [example service][1] with a performance problem that you can easily experiment with. The example service has an API which allows searching a "database" of 5000 movies. We'll fix a performance problem with it! This guide shows you the process, but for the full experience you can follow along with your own shell, browser, and IDE.
+
+## Prerequisites
+
+You'll need:
+1. [docker-compose][2]
+2. A Datadog account and [API key][3] (you don't need an application key). If you don't already have a Datadog account, [sign up for a free trial][4].
+
+## Run the Example
+
+Get the example service up and running with:
+```
+git clone https://github.com/DataDog/dd-continuous-profiler-example.git
+cd dd-continuous-profiler-example
+echo "DD_API_KEY=YOUR_API_KEY" > docker.env
```

Comment:
I should have caught this a little earlier but I don't think we should put agentless mode here as it is deprecated.